contrib/pg_tde/t/013_crash_recovery.pl

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use File::Basename;
+use Test::More;
+use lib 't';
+use pgtde;
+
+PGTDE::setup_files_dir(basename($0));
+
+my $node = PostgreSQL::Test::Cluster->new('main');
+$node->init;
+$node->append_conf('postgresql.conf', "shared_preload_libraries = 'pg_tde'");
+$node->start;
+
+PGTDE::psql($node, 'postgres', 'CREATE EXTENSION IF NOT EXISTS pg_tde;');
+PGTDE::psql($node, 'postgres',
+	"SELECT pg_tde_add_global_key_provider_file('global_keyring', '/tmp/crash_recovery.per');"
+);
+PGTDE::psql($node, 'postgres',
+	"SELECT pg_tde_set_server_key_using_global_key_provider('wal_encryption_key', 'global_keyring');"
+);
+PGTDE::psql($node, 'postgres',
+	"SELECT pg_tde_add_database_key_provider_file('db_keyring', '/tmp/crash_recovery.per');"
+);
+PGTDE::psql($node, 'postgres',
+	"SELECT pg_tde_set_key_using_database_key_provider('db_key', 'db_keyring');"
+);
+
+PGTDE::psql($node, 'postgres',
+	"CREATE TABLE test_enc (x int PRIMARY KEY) USING tde_heap;");
+PGTDE::psql($node, 'postgres', "INSERT INTO test_enc (x) VALUES (1), (2);");
+
+PGTDE::psql($node, 'postgres',
+	"CREATE TABLE test_plain (x int PRIMARY KEY) USING heap;");
+PGTDE::psql($node, 'postgres', "INSERT INTO test_plain (x) VALUES (3), (4);");
+
+PGTDE::psql($node, 'postgres', "ALTER SYSTEM SET pg_tde.wal_encrypt = 'on';");
+
+PGTDE::append_to_result_file("-- kill -9");
+$node->kill9;
+
+PGTDE::append_to_result_file("-- server start");
+$node->start;
+
+PGTDE::append_to_result_file("-- rotate wal key");
+PGTDE::psql($node, 'postgres',
+	"SELECT pg_tde_set_server_key_using_global_key_provider('wal_encryption_key_1', 'global_keyring', true);"
+);
+PGTDE::psql($node, 'postgres',
+	"SELECT pg_tde_set_key_using_database_key_provider('db_key_1', 'db_keyring');"
+);
+PGTDE::psql($node, 'postgres', "INSERT INTO test_enc (x) VALUES (3), (4);");
+PGTDE::append_to_result_file("-- kill -9");
+$node->kill9;
+PGTDE::append_to_result_file("-- server start");
+PGTDE::append_to_result_file(
+	"-- check the recovery redo and the fix in PR #217 ('Failed to verify principal key header...' error)"
```

Comment:
I would prefer not to have a reference to a PR here but rather an explanation for what we are actually testing.